inference-engine/src/readers/ir_reader/ie_ir_parser.cpp

```diff
@@ -691,12 +691,12 @@ V10Parser::V10Parser::GenericLayerParams XmlDeserializer::parseGenericParams(
         port.portId = GetIntAttr(parentNode, "id");
 
         FOREACH_CHILD(node, parentNode, "dim") {
-            size_t dim = 0;
+            int64_t dim = 0;
             const pugi::char_t* dimVal = node.child_value();
             std::stringstream ss(dimVal);
-            if (!(ss >> dim) || dim == 0) {
+            if (!(ss >> dim) || dim < 0) {
```

Comment:
Can we add a test with negative dimensions?

Reply:
It will be a completely invalid IR :) I can add a test with IR with Constant of shape [-1], but I think that it looks really redundant.

Reply:
I asked about it because it should help to cover this code.

Reply:
Done